Upgrade sqlite3 version

The current version doesn't work with Node.js 10, when trying to install it you'll see—

```
node-pre-gyp ERR! not ok 
[...]
In file included from ../node_modules/nan/nan.h:192:
../node_modules/nan/nan_maybe_43_inl.h:112:15: error: no member named 'ForceSet' in 'v8::Object'
  return obj->ForceSet(isolate->GetCurrentContext(), key, value, attribs);
[...]
8 warnings and 1 error generated.
```

`nan` is a dependency of `sqlite3`

--- a/README.md
+++ b/README.md
@@ -1,6 +1,6 @@
 # chrome-cookies-secure
 
-Extract encrypted Google Chrome cookies for a url on Mac OS X or Linux
+Extract encrypted Google Chrome cookies for a url on Mac OS X, Windows, or Linux
 
 ## Installation
 

--- a/index.js
+++ b/index.js
@@ -11,6 +11,8 @@ var sqlite3 = require('sqlite3'),
 	int = require('int'),
 	url = require('url'),
 	crypto = require('crypto'),
+	os = require('os'),
+	dpapi,
 	Cookie = tough.Cookie,
 	path,
 	ITERATIONS,
@@ -67,6 +69,12 @@ function getDerivedKey(callback) {
 
 		chromePassword = 'peanuts';
 		crypto.pbkdf2(chromePassword, SALT, ITERATIONS, KEYLENGTH, 'sha1', callback);
+
+	} else if (process.platform === 'win32') {
+
+		// On Windows, the crypto is managed entirely by the OS.  We never see the keys.
+		dpapi = require('win-dpapi');
+		callback(null, null);
 
 	}
 
@@ -242,10 +250,13 @@ const getCookies = async (uri, format, callback, profile) => {
 		path = process.env.HOME + `/.config/google-chrome/${profile}/Cookies`;
 		ITERATIONS = 1;
 	
+	} else if (process.platform === 'win32') {
+
+		path = os.homedir() + `\\AppData\\Local\\Google\\Chrome\\User Data\\${profile}\\Cookies`;
+
 	} else {
 	
-		console.error('Only Mac and Linux are supported.');
-		process.exit();
+		return callback(new Error('Only Mac, Windows, and Linux are supported.'));
 	
 	}
 
@@ -298,7 +309,13 @@ const getCookies = async (uri, format, callback, profile) => {
 
 				if (cookie.value === '' && cookie.encrypted_value.length > 0) {
 					encryptedValue = cookie.encrypted_value;
-					cookie.value = decrypt(derivedKey, encryptedValue);
+
+					if (process.platform === 'win32') {
+						cookie.value = dpapi.unprotectData(encryptedValue, null, 'CurrentUser').toString('utf-8');
+					} else {
+						cookie.value = decrypt(derivedKey, encryptedValue);
+					}
+
 					delete cookie.encrypted_value;
 				}
 

--- a/package.json
+++ b/package.json
@@ -20,7 +20,8 @@
   "license": "MIT",
   "os": [
     "darwin",
-    "linux"
+    "linux",
+    "win32"
   ],
   "repository": {
     "type": "git",
@@ -34,6 +35,7 @@
     "tough-cookie": "^2.3.4"
   },
   "optionalDependencies": {
-    "keytar": "^4.2.1"
+    "keytar": "^4.2.1",
+    "win-dpapi": "^0.1.0"
   }
 }

--- a/package.json
+++ b/package.json
@@ -30,12 +30,12 @@
   "dependencies": {
     "int": "^0.2.0",
     "request": "^2.88.0",
-    "sqlite3": "^4.0.6",
+    "sqlite3": "^4.1.1",
     "tldjs": "^1.5.1",
     "tough-cookie": "^2.3.4"
   },
   "optionalDependencies": {
-    "keytar": "^4.2.1",
+    "keytar": "^5.0.0",
     "win-dpapi": "^0.1.0"
   }
 }

